watchers/factory.go

```diff
@@ -87,14 +87,25 @@ func (f factory) NewDisruptionTargetWatcher(name string, enableObserver bool, di
 		log:            f.config.Log,
 	}
 
+	// targetObjectType can either be a pod or a node
+	var targetObjectType client.Object
```

Comment:
@nathantournant Can you update the unit tests to cover this cases?

Reply:
I will once I have something working !